pkg/controller/custom_resource_replicas.go

```diff
@@ -34,37 +34,47 @@ func (c *RolloutController) adjustStatefulSetsGroupReplicasToMirrorResource(ctx
 
 		referenceResource := fmt.Sprintf("%s/%s", referenceGVR.Resource, referenceName)
 
-		desiredReplicas := scaleObj.Spec.Replicas
-		if currentReplicas == desiredReplicas {
-			updateStatusReplicasOnReferenceResourceIfNeeded(ctx, c.logger, c.dynamicClient, sts, scaleObj, referenceGVR, referenceName, desiredReplicas)
-			cancelDelayedDownscaleIfConfigured(ctx, c.logger, sts, client, desiredReplicas)
+		referenceResourceDesiredReplicas := scaleObj.Spec.Replicas
+		if currentReplicas == referenceResourceDesiredReplicas {
+			updateStatusReplicasOnReferenceResourceIfNeeded(ctx, c.logger, c.dynamicClient, sts, scaleObj, referenceGVR, referenceName, referenceResourceDesiredReplicas)
+			cancelDelayedDownscaleIfConfigured(ctx, c.logger, sts, client, referenceResourceDesiredReplicas)
 			// No change in the number of replicas: don't log because this will be the result most of the time.
 			continue
 		}
 
 		// We're going to change number of replicas on the statefulset.
 		// If there is delayed downscale configured on the statefulset, we will first handle delay part, and only if that succeeds,
 		// continue with downscaling or upscaling.
-		if err := checkScalingDelay(ctx, c.logger, sts, client, currentReplicas, desiredReplicas); err != nil {
-			level.Warn(c.logger).Log("msg", "not scaling statefulset due to failed scaling delay check", "group", groupName, "name", sts.GetName(), "currentReplicas", currentReplicas, "desiredReplicas", desiredReplicas, "err", err)
+		var desiredReplicas int32
```

Comment:
To make code more robust from future changes, I suggest to not have this initialised to 0. We could simply do:

```suggestion
desiredReplicas, err := checkScalingDelay(ctx, c.logger, sts, client, currentReplicas, referenceResourceDesiredReplicas)
if err != nil {
   // ...
}
```